### PR TITLE
Update README.md example code

### PR DESCRIPTION
**Pull request by jsdt**
The existing code won't run without being edited, and it gives the wrong format for the parent resource.

This infers the projectId using `client.getProjectId()`, so it will work as long as default application credentials are set up.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-functions/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,16 +63,15 @@ npm install @google-cloud/functions
 // eslint-disable-next-line node/no-missing-require
 const {CloudFunctionsServiceClient} = require('@google-cloud/functions');
 
-// TODO(developer): replace with your prefered project ID.
-// const projectId = 'my-project'
-
```
Comment (sofisl) [inline README.md:68]: Thanks for submitting this review! Unfortunately, this method of authentication is just the way we do it across all of our client libraries - we do want to encourage users to authenticate beforehand.

File: README.md [hunk 2]
```diff
@@ -63,16 +63,15 @@ npm install @google-cloud/functions
...
 // Creates a client
 const client = new CloudFunctionsServiceClient();
+// TODO(developer): optionally replace with your prefered project ID.
+const projectId = await client.getProjectId();
 
-// project = 'my-project/*/locations/*' // Get the functions for a project.
+const parent = `projects/${projectId}/locations/-`; // Get all the functions for a project.
```
Comment (sofisl) [inline README.md:71]: Thank you for catching this! If you wouldn't mind leaving this as 'project', but changing the format as you described, that would be awesome!

**Comment by bcoe** [comment]
@jsdt we should make these changes in `quickstart.js` as well, the README is auto-generated from this sample file.

thanks for the contribution 👌

File: README.md (hunk 1 above)
Comment (jsdt) [reply README.md:68]: I'm not sure I understand this comment. This is using the default application credentials (which requires users to authenticate beforehand).

File: README.md (hunk 2 above)
Comment (jsdt) [reply README.md:71]: You want to me to reuse the `project` variable to change it from being the project id to the parent resource string? Or do you just want me to change `projectId` to `project`?

**Comment by bcoe** [comment]
@jsdt just a gentle nudge, would love to see this merged if you can address @sofisl's comment.

File: README.md (hunk 1 above)
Comment (sofisl) [reply README.md:68]: Right. However, we leave this section commented out so users can input their project ID if they want to.

File: README.md (hunk 2 above)
Comment (sofisl) [reply README.md:71]: Ah, here's what I would recommend: 

```
// TODO(developer): replace with your prefered project ID.
// const projectId = 'my-project'

const parent = `projects/${projectId}/locations/-`; // Get all the functions for a project.


async function listFunctions() {
  const [functions] = await client.listFunctions({
    parent,
    pageSize: page_size,
    pageToken: page_token,
  });

```

**Comment by grant** [comment]
I wasn't able to get the library working with the default README quickstart, but I saw this PR which was pretty useful.

Here's my working setup:

```ts
const {CloudFunctionsServiceClient} = require('@google-cloud/functions');

// Creates a client
const client = new CloudFunctionsServiceClient();

(async () => {
  const projectId = await client.getProjectId();
  const parent = `projects/${projectId}/locations/-`; // Get all the functions for a project.
  async function listFunctions() {
    const [functions] = await client.listFunctions({parent});
    console.info(functions);
  }
  listFunctions();  
})();
```

I assume this module is new, hence the sample isn't quite right/idiomatic (unused variables like `projectId`, use of underscores). There's also no `@google-cloud/run` library which would be nice.

**Comment by sofisl** [comment]
@jsdt and @grant, thanks for your help in debugging this sample. I'm closing this for now but using your comments for a new PR, see #55

**Pull request closed without merging**